app/assets/javascripts/application.js.erb

```diff
@@ -515,16 +465,41 @@ const onPageLoad = (callback) => {
 const replace = (oldElement, newHTMLString) => {
   const placeHolder = document.createElement('div');
   placeHolder.insertAdjacentHTML('afterBegin', newHTMLString);
-  const newElement = placeHolder.firstElementChild;
-  oldElement.replaceWith(newElement);
+  const newElements = placeHolder.childNodes.values();
+  oldElement.replaceWith(...newElements);
+}
+
+const fetchWithCSRF = (url, params) => {
+  let csrfToken = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
+  let h = new Headers();
+  h.append('X-CSRF-Token', csrfToken);
+  params = params || {};
+  params['headers'] = h;
+  return fetch(url, params);
+} 
+
+const removeExtraInput = (targetElement) => {
+  // This deletion will resovle a bug that creates an extra hidden input when rendering the comment elements.
+  const extraInput2 = targetElement.parentElement.querySelector('.comment_folder_button');
+  extraInput2.remove();
 }
 
 onPageLoad(() => {
 
+  // Global Functions
   on('click', '.markdown_help_label', (event) => {
     parentSelector(event.target, '.markdown_help_toggler').querySelector('.markdown_help').classList.toggle('display-block');
   });
 
+  // Inbox Related Funtions
+
+  on('change', '#message_hat_id', (event) => {
+    let selectedOption = event.target.options.selectedIndex;
```

Comment:
Use [`selectedOptions.first`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/selectedOptions) to avoid hassling around with indexes.

Reply:
So I had to use `event.target.selectedOptions[0]` because `.first` is a Ruby function and won't work in JS.

Reply:
...guess which language I write more of. Thanks for catching that. I see you caught some other small issues in the code suggestions I wrote off the top of my head.